static/js/publisher/builds/index.js

```diff
@@ -42,13 +44,52 @@ class Builds extends React.Component {
     }
   }
 
+  getInitialQueueTime(builds) {
+    let newQueueTime = {};
+
+    builds.forEach(build => {
+      if (build.status === "building_soon" && build.queue_time) {
+        newQueueTime[build.arch_tag] = build.queue_time;
+      }
+    });
+
+    return newQueueTime;
+  }
+
+  updateQueueTime() {
+    const { queueTime, builds } = this.state;
+    let newQueueTime = {};
+
+    builds.forEach(build => {
+      if (
+        !queueTime[build.arch_tag] &&
```

Comment:
I think this can be modified to reduce the repetition:

```
const isBuildingSoon = build.status === "building_soon" && build.queue_time;
if (isBuildingSoon) {
  newQueueTime = queueTime[build.arch_tag] ? queueTime[build.arch_tag] : build.queue_time;
}
```
I *think* that will work - you'll need to check it :D